### PR TITLE
Feature/circleci

### PR DESCRIPTION
**Pull request by tmclane**
Implements CircleCI 2.0 support for `feature/ha`.

* Removes the deprecated CircleCI 1.0 file
* Builds `zm-mailbox` submodules `native`, `common`, `soap`, `client`, and `store`
* Runs unit tests for `zm-mailbox` submodules `native`, `common`, `soap`, `client`, and `store`
* Builds a complete installer so that the zimbra-*.deb files are created - needs optimization to only build needed `zimbra-*` files
* Builds a Docker container image and pushes to the hub.docker.com repository configured

Requires the following environment variables to be defined in the CircleCI Dashboard:

* DOCKER_USER - hub.docker.com username
* DOCKER_PASS - hub.docker.com password
* APP_ENDPOINT - Repository name to push the container image into

Example output can be seen in my personal repository on hub.docker.com:

* CircleCI: https://hub.docker.com/r/tmclane/zmc-mailbox-circleci/
* Tagged builds: https://hub.docker.com/r/tmclane/zmc-mailbox-circleci/tags

Tag format:

`YEARMONTHDAY_HOURMINUTE_SHORTGITHASH`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      - run:
+          name: Running unit tests for 'zm-mailbox/store'
+          command: ant -f store/build.xml test
+
+      - run:
+          name: Preparing to build installer - checking out zm-build & zm-docker
+          command: |
+            mkdir -p "$BUILD"
+            cd "$BUILD"
+            git clone --depth=1 --branch feature/ha git@github.com:Zimbra/zm-build.git zm-build
```
[the remark this answers is not in the document]
Comment (Prashantsurana) [reply .circleci/config.yml]: Sorry it is okay as  cloning zm-build .

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            git clone --depth=1 --branch develop git@github.com:Zimbra/zm-docker.git zm-docker
```
Comment (Prashantsurana) [inline .circleci/config.yml]: Can we avoid use of zm-docker all together?
Zm-mailbox should contain entry point script from mailbox service. 
Same for config.build

File: .circleci/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tmclane) [reply .circleci/config.yml]: What do you mean by this?

File: .circleci/config.yml (hunk 2 above)
Comment (tmclane) [reply .circleci/config.yml]: Yes that can be done.
This image is intended to actually be run inside of `zm-docker` at this time and so this task will be undertaken in the future.

Comment (tmclane) [reply .circleci/config.yml]: i.e. they are tightly coupled and using the entry scripts from zm-docker is necessary to enable it to run as expected in that environment.

File: .circleci/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tmclane) [reply .circleci/config.yml]: Yes I could use CIRCLE_BRANCH if that is what you are referring to. Although in this case the zm-build repository may or may not be the same branch as what is being processed by CircleCI.

**Pull request merged**